CenterForOpenScience/ember-osf-preprints
# [IN-102][Preprints] Enable user search to use additional profile information

## Purpose

Enables search by other fields for users, like orcids, etc

## Summary of Changes/Side Effects

- Use new model added to ember-osf, search-users. 
- The query is lucene (afaict) which means it won't do partial matching without modification (e.g., searching alex does not return alexander). The previous endpoint did do partial matching. So to hack this together, instead of telling people it is lucene, this pre- and post-pends `*` to the query to allow the loose matching. Confirmed this with Sara, will demo it to be sure. Alternative is to tell people it is lucene, but that is less than ideal.

## Testing Notes

1. Test various fields and combos to make sure things that used to work (fullname, middle name, first name, last name) all still work properly.
2. Try various additional profile information to see if results match what you'd expect.
3. Try some partial and full matches to make sure all of them work (alex for alexander; lex for alex; alex for alex, etc)
4. Make sure pagination works properly (I didn't mess with it, but who knows).

## Ticket

https://openscience.atlassian.net/browse/IN-102

## Notes for Reviewer

Nada nada limonada 

# Reviewer Checklist
